cmake/fpu-for-gcc-m-cpu.cmake

```diff
@@ -1,5 +1,9 @@
 # Defines a mapping from GCC_M_CPU to FPU
 
 set(FPU_FOR_cortex-m4      fpv4-sp-d16)
+if(CONFIG_CPU_HAS_FPU_DOUBLE_PRECISION)
 set(FPU_FOR_cortex-m7      fpv5-d16)
+else()
+set(FPU_FOR_cortex-m7      fpv5-sp-d16)
+endif()
```

Comment:
Let's make it more obvious from the code that it is impossible to have double precision on m4 and m33.

So,

```
if(CONFIG_CPU_HAS_FPU_DOUBLE_PRECISION)
  set(FPU_FOR_cortex-m4      does_not_exist)
  set(FPU_FOR_cortex-m7      fpv5-d16)
  set(FPU_FOR_cortex-m33     does_not_exist)
else()
  set(FPU_FOR_cortex-m4      fpv4-sp-d16)
  set(FPU_FOR_cortex-m7      fpv5-sp-d16)
  set(FPU_FOR_cortex-m33     fpv5-sp-d16)
endif()


```

alternatively, we could be agnostic to which platforms have which precision by doing:

```
if(CONFIG_CPU_HAS_FPU_DOUBLE_PRECISION)
  set(PRECISION_TOKEN sp-)
else()
  set(PRECISION_TOKEN)
endif()

set(FPU_FOR_cortex-m4      fpv4-${PRECISION_TOKEN}d16)
set(FPU_FOR_cortex-m7      fpv5-${PRECISION_TOKEN}d16)
set(FPU_FOR_cortex-m33     fpv5-${PRECISION_TOKEN}d16)

```

as it is currently written, it is not possible to determine from the code if we simply have not added support for double precision on more platforms than m7, or if m7 is the only platform that can have double precision. But in the two proposals above this question can be answered from reading the code.

Reply:
I like your second approach

Reply:
@SebastianBoe I went with your second approach. Actually the "revert" option, since the sp- infix goes if we are _not_ on double precision.

pls, revisit, and thanks for looking at it

Reply:
haha, oops